### PR TITLE
write 93 unit tests

### LINKED ISSUE #93: (SP: 1) Write unit test for "PopupDialog" component
### Component unit test
Unit test for "**PopupDialog**" component
Scenaries descriptions:
- [ ] Content text should be shown
- [ ] Imitate click on the close button and popup should be closed
- [ ] Imitate click with delay and popup should be closed

[Link to component](https://github.com/Space2Study-UA-1125/frontEnd/blob/develop/src/components/popup-dialog/PopupDialog.jsx)

Current coverage:
Image

### PR DESCRIPTION
**Pull request by ashfox21**
I mock 2 functions that the component gets in props. One closes the window and the other closes after some time.
I checked the rendering of the component and the content passed to it for rendering.
I checked the call to the close function when the close button is clicked.
Checked the call to the delayed close function when the mouse is removed from the component. 
I also checked tests against false positives by changing the source code of the component.
Coverage :
image

### CODE REVIEW

**Review by Wiret02 — APPROVED** [review]

File: tests/unit/components/popup-dialog/PopupDialog.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  beforeEach(() => {
+    render(<PopupDialog {...props} />)
+  })
+
+  it('Component renders correctly', () => {
+    const contentElement = screen.getByText('TestContent')
+    expect(contentElement).toBeInTheDocument()
+  })
+
+  it('Dialog closes when you click on the close button', async () => {
```
Comment (iva-stasia) [inline tests/unit/components/popup-dialog/PopupDialog.spec.jsx]: There is nothing asynchronous inside, so you don't need `async/await`. `userEvent.click()` is synchronous.

File: tests/unit/components/popup-dialog/PopupDialog.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const closeElement = screen.getByTestId('CloseIcon')
+
+    await userEvent.click(closeElement)
+
+    expect(props.closeModal).toHaveBeenCalled()
+  })
+
+  it('Dialog closes with 5 seconds of inactivity', async () => {
```
Comment (iva-stasia) [inline tests/unit/components/popup-dialog/PopupDialog.spec.jsx]: This `async` is useless here, all actions inside are synchronous.

File: tests/unit/components/popup-dialog/PopupDialog.spec.jsx
Comment (iva-stasia) [inline tests/unit/components/popup-dialog/PopupDialog.spec.jsx:1]: Please pay attention to my comments in your PR #134. Some of them work here as well.

**Review by iva-stasia — CHANGES_REQUESTED** [review]

File: tests/unit/components/popup-dialog/PopupDialog.spec.jsx (hunk 1 above)
Comment (ashfox21) [reply tests/unit/components/popup-dialog/PopupDialog.spec.jsx]: This click starts an asynchronous function inside the component, I can't spy this function, without asynchrony in this test.

File: tests/unit/components/popup-dialog/PopupDialog.spec.jsx
Comment (ashfox21) [reply tests/unit/components/popup-dialog/PopupDialog.spec.jsx:1]: I changed the test descriptions, is that what you meant?
image
image

File: tests/unit/components/popup-dialog/PopupDialog.spec.jsx (hunk 2 above)
Comment (ashfox21) [reply tests/unit/components/popup-dialog/PopupDialog.spec.jsx]: Understood, I'll remove that part

**Review by OliaHrytsak — APPROVED** [review]

**Review by holus-bolus — APPROVED** [review]

**Review by VictoriaRodik — APPROVED** [review]

**Review by iva-stasia — APPROVED** [review]

**Review by online-marathon — APPROVED** [review]
It is a good example of asynchronous tests. `advanceTimersByTime` looks well here.

**Pull request merged**